api/graphql/reservations/reservation_mutations.py

```diff
@@ -114,3 +117,21 @@ class Meta:
         model_operations = ["update"]
         lookup_field = "pk"
         serializer_class = ReservationUnitAdjustTimeSerializer
+
+
+class ReservationDeleteMutation(AuthDeleteMutation, ClientIDMutation):
+    permission_classes = (ReservationPermission,)
+    model = Reservation
+
+    @classmethod
+    def validate(self, root, info, **input):
+        reservation = get_object_or_404(Reservation, pk=input.get("pk", None))
+        if reservation.state not in (
+            ReservationState.CREATED,
+            ReservationState.WAITING_FOR_PAYMENT,
+        ):
+            raise ValidationError(
```

Comment:
We could use `ValidationErrorWithCode` here

Reply:
Its kinda decision to not use it here since it would be quite a change and would need a refactor and change other deletions to use it as well. Which sounds awesome, but would take a bit of more time so decided not to since we have the other ticket relating to error codes to root.